src/ir/auth_logic/ast.h

```diff
@@ -0,0 +1,182 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+
+// This file describes AST nodes for authorization logic
+
+#ifndef SRC_IR_AUTH_LOGIC_AST_H_
+#define SRC_IR_AUTH_LOGIC_AST_H_
+
+#include <string>
+#include <vector>
+#include <variant>
+
+namespace raksha::ir::auth_logic {
+
+class AstPrincipal {
+  public:
+    AstPrincipal(std::string name) : name_(std::move(name)) {}
```

Comment:
Do we have any restrictions on what characters/format are allowed in a principal's `name`? If so, this constructor might be a good place to `CHECK` that they are honored, as it will allows us to trust that those constraints are in place for the rest of the object's lifetime.

Reply:
The parser should already check that the principal names are in a particular format, so we shouldn't need to replicate that here, I think.